content/docs/io/overview.md

```diff
@@ -6,7 +6,119 @@ menu:
     parent: io
 ---
 
-This page has not been worked on yet. If you'd like to contribute visit the [doc-push]
-repo.
+The Rust standard library provides support for networking and I/O, such
+as TCP connections, UDP sockets, reading from and writing to files, etc.
+However, those operations are all synchronous, or _blocking_, meaning
+that when you call them, the current thread may stop executing and go to
+sleep until it is unblocked. For example, `read` will block until there
+is data to read. In the world of futures, that behavior is unfortunate,
+since we would like to continue executing other futures we may have
+while waiting for the I/O to complete.
 
-[doc-push]: https://github.com/tokio-rs/doc-push
+To enable this, Tokio provides _non-blocking_ versions of many standard
+library I/O resources, such as [file operations] and [TCP], [UDP], and
+[Unix] sockets. They return futures for long-running operations (like
+accepting a new TCP connection), and implement non-blocking variants of
+`std::io::Read` and `std::io::Write` called `AsyncRead` and
+`AsyncWrite`.
+
+Non-blocking reads and writes do not block if, for example, there is no
+more data available. Instead, they return immediately with a
+`WouldBlock` error, along with a promise (like `Future::poll`) that they
+have arranged for the current task to be woken up when they can later
+make progress, such as when a network packet arrives. 
+
+By using the non-blocking Tokio I/O types, a future that performs I/O
+no longer blocks execution of other futures if the I/O they wish to
+perform cannot be performed immediately. Instead, it simply returns
+`NotReady`, and relies on a task notification to cause `poll` to be
```

Comment:
Is it to early to specifically call this "task notification" or should this be more hand-wavy?

Reply:
The previous section (working with futures) will have talked about this (well, once that is written).